netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -221,6 +229,7 @@ public void shutdown() {
     notifyShutdown(null);
     // Notifying of termination is automatically done when the channel closes.
     if (channel != null && channel.isOpen()) {
+      // TODO(jhump): send a GOAWAY frame? (that's what okhttp impl does)
```

Comment:
Closing the channel will close the transport, which will automatically send a GOAWAY frame. This is all handled by Netty.